src/content_sao.cpp

```diff
@@ -1102,13 +1102,12 @@ int PlayerSAO::punch(v3f dir,
 
 	if (!damage_handled) {
 		setHP(getHP() - hitparams.hp);
-	} else { // override client prediction
-		if (puncher->getType() == ACTIVEOBJECT_TYPE_PLAYER) {
-			std::string str = gob_cmd_punched(0, getHP());
-			// create message and add to list
-			ActiveObjectMessage aom(getId(), true, str);
-			m_messages_out.push(aom);
-		}
+	} else if (puncher && puncher->getType() == ACTIVEOBJECT_TYPE_PLAYER) {
```

Comment:
Since this code never caused a problem before, is it logically impossible for puncher to == NULL here?  If so, maybe it ought to be an assertion instead.